torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -243,17 +243,23 @@ xla::XlaOp CreateMatMul(const xla::XlaOp& lhs, const xla::XlaOp& rhs) {
               << rhs_shape << ")";
 }
 
+xla::XlaOp BuildBernoulli(const xla::XlaOp& probability,
+                          const xla::Shape& shape) {
+  xla::Shape probability_shape = XlaHelpers::ShapeOfXlaOp(probability);
```

Comment:
probablity_shape is always shape.

Reply:
What is your suggestion here?

Reply:
If your claim is that `shape` and `probability_shape` are always the same, I don't think they are. They can differ in element type: for example, `XLATensor::bernoulli_` preserves the shape of input and takes the probability from a different tensor. They are only guaranteed to be the same for `XLATensor::bernoulli`.

Reply:
I am not in love with the fact that in many places we are coming here, we know that probability_shape is shape, and we issue a call to XlaHelpers::ShapeOfXlaOp to get it.
Give that we are generating a noise from 0 to 1, I think we can pun to F32 as noise type (which is only used to build the mask).
At that point we need no probability_shape anymore as all the paths leading here have same dimensions.

Reply:
Making the noise F32 sounds ok, I'll change it.

Reply:
Not sure what you meant here. We need a shape for `xla::RngUniform` and `shape` can have any element type. Would you rather create `probability_shape` from `shape`, with `F32` as an element type? That'd avoid querying it, but I don't understand why querying it is bad.

Reply:
OK, leave it as is.